baselines/FedStar/FedStar/server.py

```diff
@@ -0,0 +1,219 @@
+        self.min_sample_size = flwr_min_sample_size
+        self.min_num_clients = flwr_min_num_clients
+        self.rounds = flwr_rounds
+        # Model Parameters
+        self.num_classes = model_num_classes
+        self.lr = model_lr
+        self.batch_size = model_batch_size
+        self.epochs = model_epochs
+        self.verbose = model_verbose
+        self.ds_test = model_ds_test
+        # Local Variables Counters and Variables
+        self.current_round = 0
+        self.final_accuracy = 0.0
+        self.round_time = time.time()
+        self.strategy = flwr.server.strategy.FedAvg(
+            fraction_fit=self.sample_fraction,
+            min_fit_clients=self.min_sample_size,
+            min_available_clients=self.min_num_clients,
+            on_fit_config_fn=self.get_on_fit_config_fn(),
+            fraction_evaluate=0,
+            min_evaluate_clients=0,
+            on_evaluate_config_fn=None,
+            evaluate_fn=self.get_eval_fn(ds_test=self.ds_test),
+            accept_failures=True,
+        )
+        self.client_manager = flwr.server.client_manager.SimpleClientManager()
+        tf.keras.backend.clear_session()
+
+    def server_start(self, server_address):
+        print("|"*50)
+        print(server_address)
+        flwr.server.start_server(
+            server_address="0.0.0.0:8080",
+            server=flwr.server.Server(
+                client_manager=self.client_manager, strategy=self.strategy
+            ),
+            config=flwr.server.ServerConfig(num_rounds=self.rounds),
+            strategy=self.strategy,
+            grpc_max_message_length=GRPC_MAX_MESSAGE_LENGTH,
+        )
+
+    def get_on_fit_config_fn(self):
+        def fit_config(
+            rounds=self.rounds,
+            epochs=self.epochs,
+            batch_size=self.batch_size,
+            learning_rate=self.lr,
+        ):
+            if self.current_round != 1:
+                print(
+                    "\nTraining round completed in "
+                    + str(round(time.time() - self.round_time, 2))
+                    + " seconds."
+                )
+            print(
+                "\n%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n"
+                + "               Server started "
+                + str(self.current_round)
+                + "th round of training.\n"
+                + "%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n"
+            )
+            # Update round start time
+            self.round_time = time.time()
+            return {
+                "rounds": str(self.rounds),
+                "c_round": str(self.current_round),
+                "epochs": str(epochs),
+                "batch_size": str(batch_size),
+                "learning_rate": str(learning_rate),
+            }
+
+        return fit_config
+
+    def get_eval_fn(self, ds_test):
+        import tensorflow as tf
```

Comment:
```suggestion
```

Reply:
do we need to re-import it here?

Reply:
Thanks for the suggestion, I have removed the imports.